packages/datetime/src/dateInput.tsx

```diff
@@ -355,7 +366,7 @@ export class DateInput extends AbstractComponent<IDateInputProps, IDateInputStat
 
     private handleInputBlur = (e: React.FocusEvent<HTMLInputElement>) => {
         const valueString = this.state.valueString;
-        const value = moment(valueString, this.props.format);
+        const value = moment(valueString, this.props.format, this.props.locale);
```

Comment:
i feel like a `createMoment(valueString)` wrapper would be useful, in case we ever want to change it again in the future. this change appears 3 times in this PR.